grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -940,6 +912,10 @@ public boolean fixDuplicateCastings(String index, Set<ConceptId> castingVertexId
             //Remove Redundant Relations
             duplicateRelations.forEach(relation -> ((ConceptImpl) relation).deleteNode());
 
+            //Restore the index
+            String newIndex = mainCasting.getIndex();
```

Comment:
This is because deleting the duplicate castings deleted the index. This puts the index back on. Without this we end up with corrupted indices. This was a particularly fun one to solve